### ISSUE TITLE
Recommended running process

### ISSUE THREAD

**Issue by scn0901** [MaartenGr/BERTopic#1176]
Hello, I want to use bertopic for topic modeling in my research. When I studied the code examples provided by bertopic and briefly read the full documentation, I understood the various functions of bertopic. However, I'm not very clear about the complete running process.

I think that using bertopic for complete topic modeling (rather than running a simple demo) mainly includes the following steps: model training, visualization, topic reduction, topic representation adjustment, hyperparameter fine-tuning, outlier reduction, stop words removal, model saving.

My questions are: in what order do you recommend performing these steps? And what steps do you think are necessary, but I didn't list them above? Where should they be located in the running process?

**Comment by MaartenGr** [comment]
> My questions are: in what order do you recommend performing these steps?

Unfortunately, there is not one order for running these steps. It will highly depend on your use case and the things that are important and vital to it. For example, a hierarchical visualization might be more on less important depending on whether any hierarchy is expected. Running this visualization for the sake of running it is not ideal. 

Having said that, when you created the model, users typically try to understand what the model has produced. This can be done with visualizations, topic descriptions, representative documents, etc. Then, you might continue with topic representations or go back to fine-tuning the clustering process. Here, some steps might not be as important, like outlier reduction if you are using k-Means instead of HDBSCAN. 

In other words, make sure to choose and customize these steps based on what you want to achieve instead of coming up with a "default" series of steps. 

> And what steps do you think are necessary, but I didn't list them above?

BERTopic has gotten quite big over the years, functionality-wise, so there are a number of interesting features to use. In practice, I would start with fitting the model using the default pipeline and try to explore the different functionalities to see what is useful to your specific use case. If that means going back to the model to fine-tune it, then that is a good next step for you. Note that this is often an iterative process and it should be regarded as such.

**Comment by MaartenGr** [comment]
Closing this due to inactivity. Let me know if I need to re-open the issue!
